harvester_e2e_tests/integration/test_images.py

```diff
@@ -0,0 +1,375 @@
+
+        assert 200 == resp.status_code, (
+            f"Failed to upload fake image with error:{resp.status_code}, {resp.content}"
+        )
+
+        get_image(api_client, unique_name)
+        delete_image(api_client, unique_name, wait_timeout)
+
+        resp = api_client.images.create_by_file(unique_name, fake_image_file)
+
+        assert 200 == resp.status_code, (
+            f"Failed to upload fake image with error:{resp.status_code}, {resp.content}"
+        )
+
+        get_image(api_client, unique_name)
+        delete_image(api_client, unique_name, wait_timeout)
+
+    @pytest.mark.images_p1
+    def test_create_invalid_file(self, api_client, unique_name, fake_invalid_image_file):
+        """
+        Test create upload image from invalid file type
+
+        Steps:
+        1. Try to upload invalid image file to images page, something like dmg, or tar.gzps
+        2. Check should get an error
+        """
+
+        resp = api_client.images.create_by_file(unique_name, fake_invalid_image_file)
+
+        assert 500 == resp.status_code, (
+            f"File size correct, it's a multiple of 512 bytes:{resp.status_code}, {resp.content}"
+        )
+
+    @pytest.mark.images_p1
+    @pytest.mark.dependency(name="edit_image_in_use", depends=["create_image_url"])
+    def test_edit_image_in_use(self, api_client, request, unique_name, qcow2_name):
+        """
+        Test can edit image which already in use
+
+        Steps:
+        1. Check the image created from URL exists
+        2. Create a VM and use the existing image created from url
+        3. Update the image labels and description
+        4. Check can change the image content
+        """
+
+        wait_timeout = request.config.getoption('--wait-timeout')
+
+        get_image(api_client, qcow2_name)
+
+        spec = api_client.vms.Spec(1, 1)
+        spec.add_image(qcow2_name, "default/" + qcow2_name)
+        code, data = api_client.vms.create(unique_name, spec)
+
+        assert 201 == code, (f"Failed to create vm with error: {code}, {data}")
+
+        # Check VM start in running state
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+
+        while endtime > datetime.now():
+            code, data = api_client.vms.get(unique_name)
+            vm_fields = data['metadata']['fields']
+
+            assert 200 == code, (code, data)
+            if vm_fields[2] == 'Running':
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"Failed to create VM {unique_name} in Running status, exceed 10 minutes\n"
+                f"Still got {code} with {data}"
+            )
+
+        updates = {
+            "labels": {
+                "usage-label": "yes"
+            },
+            "annotations": {
+                "field.cattle.io/description": 'edit image in use'
+            },
+
+        }
+
+        update_by_input(api_client, qcow2_name, updates)
```

Comment:
If the function will only be called once, we could just place codes in here.

and we should remove the VM/volume/image after test.